### PR TITLE
scala docs

### PR DESCRIPTION
**Pull request by qiuxin2012**

### CODE REVIEW

File: readthedocs/source/doc/UserGuide/scala.md [hunk 1]
```diff
@@ -6,12 +6,153 @@
...
+Currently, Analytics Zoo releases are hosted on maven central; here's an example to add the Analytics Zoo dependency to your own project:
+```xml
+<dependency>
+    <groupId>com.intel.analytics.zoo</groupId>
+    <artifactId>analytics-zoo-[bigdl_0.10.0|bigdl_0.12.1]-[spark_2.1.1|spark_2.2.1|spark_2.3.1|spark_2.4.3|spark_3.0.0]</artifactId>
+    <version>${ANALYTICS_ZOO_VERSION}</version>
+</dependency>
+```
+You can find the latest ANALYTICS_ZOO_VERSION [here](https://search.maven.org/search?q=analytics-zoo-bigdl).   
+Please note: BIGDL VERSION is bound to ANALYTICS_ZOO_VERSION, you may need to change BIGDL VERSION when you update ANALYTICS_ZOO_VERSION.
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:18]: no need to list all the options; just use 0.9.0

File: readthedocs/source/doc/UserGuide/scala.md [hunk 2]
```diff
@@ -6,12 +6,153 @@
...
+
+SBT developers can use
+```sbt
+libraryDependencies += "com.intel.analytics.zoo" % "analytics-zoo-[bigdl_0.10.0|bigdl_0.12.1]-[spark_2.1.1|spark_2.2.1|spark_2.3.1|spark_2.4.3|spark_3.0.0]" % "${ANALYTICS_ZOO_VERSION}"
+```
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:23]: just use 0.9.0

File: readthedocs/source/doc/UserGuide/scala.md [hunk 3]
```diff
@@ -6,12 +6,153 @@
...
 
 #### **1.2 Nightly Build**
 
-<TODO: Maven & SBT>
+Currently, Analytics Zoo development version is hosted on [SonaType](https://oss.sonatype.org/content/groups/public/com/intel/analytics/zoo/).
+
+To link your application with the latest Analytics Zoo development version, you should add some dependencies like [official releases](#11-official-release), but set `${ANALYTICS_ZOO_VERSION}` to latest version, and add below repository to your pom.xml.
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:29]: just use `0.10.0-SNAPSHOT`; seems we have not published it though?

File: readthedocs/source/doc/UserGuide/scala.md [hunk 4]
```diff
@@ -6,12 +6,153 @@
...
+
+Before running Analytics Zoo, Environment Variables **ANALYTICS_ZOO_HOME** and **SPARK_HOME** should be set:
+
+* If you download Analytics Zoo from the [Release Page](../release-download.md)
+```bash
+export SPARK_HOME=folder path where you extract the spark package
+export ANALYTICS_ZOO_HOME=folder path where you extract the analytics zoo package
+```
+
+* If you build Analytics Zoo by yourself
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:64]: this should be documented in the developer guide, not here

File: readthedocs/source/doc/UserGuide/scala.md [hunk 5]
```diff
@@ -6,12 +6,153 @@
...
+```
 
 #### **1.3 Download Pre-Built Package**
 
+You can download the Analytics Zoo release and nightly build from the [Release Page](../release.md)
+
 ### **2. Run**
+
+Before running Analytics Zoo, Environment Variables **ANALYTICS_ZOO_HOME** and **SPARK_HOME** should be set:
+
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:57]: what if I'm using maven or sbt?

File: readthedocs/source/doc/UserGuide/scala.md [hunk 6]
```diff
@@ -6,12 +6,153 @@
...
+Type :help for more information.
+```
+
+Now you'll be able to play with Analytics Zoo API's.
+For instance, to load a pre-trained object detection model, you may try below code:
+```scala
+scala> import com.intel.analytics.zoo.models.image.objectdetection.ObjectDetector
+import com.intel.analytics.zoo.models.image.objectdetection.ObjectDetector
+
+scala> ObjectDetector.loadModel[Float](params.modelPath)
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:96]: use a simpler example that does not need to download or prepare the model

File: readthedocs/source/doc/UserGuide/scala.md [hunk 7]
```diff
@@ -6,12 +6,153 @@
...
+```
+
+---
+
+#### **2.2 Run as a Spark Program**
+You can run a analytics zoo program, e.g., the [Object Detection](https://github.com/intel-analytics/analytics-zoo/tree/master/zoo/src/main/scala/com/intel/analytics/zoo/examples/objectdetection), as a standard Spark program (running in either local mode or cluster mode) as follows:
+
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:103]: is there a simpler example?

File: readthedocs/source/doc/UserGuide/scala.md (hunk 3 above)
Comment (qiuxin2012) [reply readthedocs/source/doc/UserGuide/scala.md:29]: it's hidden due to `Too many results, please refine the search condition.`

**Comment by jason-dai** [comment]
We should reorganize the document as follows:
1) Try Analytics Zoo Examples
    1.1) Download and config
    1.2) Use Spark interactive shell
    1.3) Run Analytics Zoo examples
2) Build Analytics Zoo Applications
    2.1) Add Analytics Zoo dependency
    2.2) Build a Scala project

File: readthedocs/source/doc/UserGuide/scala.md [hunk 8]
```diff
@@ -6,12 +6,214 @@
...
+* If you build Analytics Zoo by yourself
+```bash
+export SPARK_HOME=folder path where you extract the spark package
+export ANALYTICS_ZOO_HOME=the dist folder generated by the build process, which is under the top level of the source folder
+```
+
+#### **2.2.1 Run with Spark interactive shell**
+
+You can try Analytics Zoo easily using the Spark interactive shell. Run below command to start spark shell with Analytics Zoo support:
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:80]: we can make it simple; maybe just run `initNNContext` to verify the environment

File: readthedocs/source/doc/UserGuide/scala.md [hunk 9]
```diff
@@ -6,12 +6,214 @@
...
 #### **1.3 Download Pre-Built Package**
 
+You can download the Analytics Zoo release and nightly build from the [Release Page](../release.md)
+
+---
 ### **2. Run**
+
+
+### **2.1 Run user application**
+Here is a [simple MLP example](https://github.com/qiuxin2012/SimpleMlp) to show you how to intergate analytics zoo to your project with maven or sbt, and how to build and run your own deep learning application in IDEA and spark-submit.
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:59]: Move the project to https://github.com/intel-analytics/zoo-tutorials

File: readthedocs/source/doc/UserGuide/scala.md (hunk 8 above)
Comment (qiuxin2012) [reply readthedocs/source/doc/UserGuide/scala.md:17]: Just start the spark-shell and run a initNNContext? Delete the codes behind the initNNContext?

Comment (jason-dai) [reply readthedocs/source/doc/UserGuide/scala.md:17]: Yes, just run a simple test to verify the environment

File: readthedocs/source/doc/UserGuide/scala.md [hunk 10]
```diff
@@ -2,16 +2,150 @@
...
+
+SBT developers can use
+```sbt
+resolvers += "ossrh repository" at "https://oss.sonatype.org/content/repositories/snapshots/"
+```
+
+
+#### **2.2 Build a Scala project**
+To enable Analytics Zoo in project, you should add Analytics Zoo to your project's dependencies using maven or sbt. 
+Here is a [simple MLP example](https://github.com/qiuxin2012/SimpleMlp) to show you how to use analytics zoo to build your own deep learningown deep learning project with maven or sbt, and how to build and run your application in IDEA and spark-submit.
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/scala.md:150]: Move the example project to `zoo-tutorials` repo

**Pull request merged**